### PR TITLE
Ownerless groups

### PR DESCRIPTION
**Pull request by IsaacG**
Make the owner player arg optional for AddGroup()

### CODE REVIEW

File: backend/api_calls.py [hunk 1]
```diff
@@ -261,17 +261,21 @@ def AddGroup(request, firebase):
   """
   results = []
   valid_args = ['!groupId', 'gameId', 'ownerPlayerId', 'allegianceFilter']
```
Comment (DolceTriade) [inline backend/api_calls.py:263]: To avoid duplication, you could just append ownerPlayerId to valid_args once you've created required_args.

Comment (DolceTriade) [reply backend/api_calls.py:263]: blah, that would just require more special casing to handle the !groupId. Feel free to ignore.

File: backend/api_calls.py [hunk 2]
```diff
@@ -261,17 +261,21 @@ def AddGroup(request, firebase):
...
-  required_args = list(valid_args)
+  required_args = ['groupId', 'gameId', 'allegianceFilter']
   required_args.extend(['autoAdd', 'autoRemove', 'membersCanAdd', 'membersCanRemove'])
   ValidateInputs(request, firebase, required_args, valid_args)
 
-  put_args = list(required_args)
-  put_args.remove('!groupId')
+  put_args = set(required_args) - set(['groupId'])
   group_data = {k: request[k] for k in put_args}
```
Comment (DolceTriade) [inline backend/api_calls.py:269]: Can't you do 
```
group_data = {k: request[k] for k in put_args if k in request}
```

This way, you can remove a lot of the duplication above and the if statement below.

Comment (IsaacG) [reply backend/api_calls.py:269]: That could include extraneous args that the user may pass in. We're not checking for additional un-required args.

Comment (DolceTriade) [reply backend/api_calls.py:269]: How so? You only add the args in put_args, but don't add them if they don't exist in request?

**Pull request merged**